### PR TITLE
Ensure rooms are recalculated on re-invites

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/element-web/issues/22106

**Reviewer**: This has two commits: a failing test and the fix. This is to demonstrate that the test actually works, because I don't exactly have confidence in it myself.

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Ensure rooms are recalculated on re-invites ([\#2374](https://github.com/matrix-org/matrix-js-sdk/pull/2374)). Fixes vector-im/element-web#22106.

### CODE REVIEW

File: spec/integ/matrix-client-syncing.spec.js [hunk 1]
```diff
@@ -1,5 +1,4 @@
-import { MatrixEvent } from "../../src/models/event";
-import { EventTimeline } from "../../src/models/event-timeline";
+import { EventTimeline, MatrixEvent, RoomEvent } from "../../src";
```
Comment (t3chguy) [inline spec/integ/matrix-client-syncing.spec.js:1]: Missing copyright

File: src/sync.ts [hunk 2]
```diff
@@ -1166,6 +1166,9 @@ export class SyncApi {
                 room.recalculate();
                 client.store.storeRoom(room);
                 client.emit(ClientEvent.Room, room);
+            } else {
+                // Update room state for invite->reject->invite cycles
+                room.recalculate();
```
Comment (t3chguy) [inline src/sync.ts:1171]: Can we not just move the recalculate out to before the `if`?

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

**Comment by turt2live** [comment]
Slightly worried about this change: if people notice performance regressions, please flag.

**Pull request merged**